### PR TITLE
Add feature to allow using encrypted PVCs for VshnPostgreSQL

### PR DESCRIPTION
**Pull request by TheBigLee**
## Checklist

- [x] The PR has a meaningful title. It will be used to auto generate the
      changelog.
      The PR has a meaningful description that sums up the change. It will be
      linked in the changelog.
- [x] PR contains a single logical change (to build a better changelog).
- [ ] Update the documentation. *Don't forget to generate the CRD API!*
- [x] Categorize the PR by adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog.
- [ ] Link this PR to related issues or PRs.

### CODE REVIEW

**Comment by TheBigLee** [comment]
I currently don't know how I can avoid the creation of the secret if the instance has encryption disabled.
Any ideas here?

**Comment by mhutter** [comment]
Somewhat relevant: cloudscale-ch/csi-cloudscale#55

**Comment by Kidswiss** [comment]
> I currently don't know how I can avoid the creation of the secret if the instance has encryption disabled.
> Any ideas here?

You need to use composition functions to make conditionals work. We're currently working on getting things ready here: https://github.com/vshn/appcat-comp-functions

File: component/vshn_postgres.jsonnet [hunk 1]
```diff
@@ -392,6 +392,21 @@ local sgCluster = {
...
+          },
+        },
+  patches: [
+    comp.ToCompositeFieldPath('status.conditions', 'status.secretConditions'),
+    comp.FromCompositeFieldPathWithTransformPrefix('metadata.labels[crossplane.io/composite]', 'metadata.name', 'secret-observer'),
+    comp.FromCompositeFieldPathWithTransformPrefix('metadata.labels[crossplane.io/composite]', 'spec.forProvider.manifest.metadata.namespace', 'vshn-postgresql'),
+    comp.CombineCompositeFromTwoFieldPaths('metadata.labels[crossplane.io/composite]', 'metadata.labels[crossplane.io/composite]', 'spec.forProvider.manifest.metadata.name', '%s-data-%s-0-luks-key'),
+  ],
+};
+
```
Comment (glrf) [inline component/vshn_postgres.jsonnet:409]: Isn't this secret completely managed by the composition function? So we don't really need this right?

**Review by glrf — APPROVED** [review]
I think the secret is unnecessary but overall LGTM

File: component/vshn_postgres.jsonnet (hunk 1 above)
Comment (TheBigLee) [reply component/vshn_postgres.jsonnet:409]: yes that is true. Removed it again.

**Pull request merged**